### PR TITLE
Adding Id parameter to performence extension

### PR DESCRIPTION
**Pull request by SaranganJ**
## Purpose

If we initialize the performance extension at multiple places in the Siddhi code (In normal deployment mode) then each performance call will write the metric to the same file.To differentiate multiple performance calls we need to insert an id parameter for the extension
Refer issue #853

## Approach

 Include an additional parameter to the extension where the id should be specified.Thus the output file created will have that id in it.So we can differentiate the performance metrics using  output file name.

### CODE REVIEW

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java [hunk 1]
```diff
@@ -58,16 +57,38 @@
...
+                        description = "This value used to find the sending timestamp from client",
+                        type = {DataType.LONG}),
+
+                @Parameter(name = "type",
+                        description = "This value used to specify which metric value " +
+                                "you are going to exract(\"throughput\" " +
+                                "or \"latency\" or \"both\" )",
+                        type = {DataType.STRING}),
+
+                @Parameter(name = "windowsize",
```
Comment (Niveathika92) [inline modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: change it to : window.size

Comment (SaranganJ) [reply modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Fixed https://github.com/wso2/product-sp/pull/880/commits/9ee8cbb4f4af6e57527b1f8b83363f2fc400a552

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java [hunk 2]
```diff
@@ -219,26 +243,26 @@ private static int getLogFileSequenceNumber() {
...
+            if (!(attributeExpressionExecutors[0].getReturnType() == Attribute.Type.LONG)) {
                 throw new SiddhiAppValidationException("iijTimestamp is expected to be long but found "
                         + attributeExpressionExecutors[0].getReturnType());
             }
 
             if (!(attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor)) {
-                throw new SiddhiAppValidationException("second parameter has to be constant but " +
-                        "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
+                throw new SiddhiAppValidationException("Type has to be constant(\"throughput\"or " +
+                        "\"latency\" or \"both\") but " + "found " +
```
Comment (miyurud) [inline modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: You could change this to

"\"latency\" or \"both\") but  found " +

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java [hunk 3]
```diff
@@ -219,26 +243,26 @@ private static int getLogFileSequenceNumber() {
...
-            } else {
+            if (!(attributeExpressionExecutors[0].getReturnType() == Attribute.Type.LONG)) {
                 throw new SiddhiAppValidationException("iijTimestamp is expected to be long but found "
                         + attributeExpressionExecutors[0].getReturnType());
             }
 
             if (!(attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor)) {
-                throw new SiddhiAppValidationException("second parameter has to be constant but " +
-                        "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
+                throw new SiddhiAppValidationException("Type has to be constant(\"throughput\"or " +
```
Comment (miyurud) [inline modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Need a space

throw new SiddhiAppValidationException("Type has to be constant(\"throughput\" or " +

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java [hunk 4]
```diff
@@ -288,7 +334,7 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
 
             default:
                 log.error("executionType should be either throughput or latency or both "
-                                  + "but found" + " " + executionType);
+                        + "but found" + " " + executionType);
```
Comment (miyurud) [inline modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: + "but found" + " " can be changed to + "but found " +

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java [hunk 5]
```diff
@@ -249,18 +273,40 @@ private static int getLogFileSequenceNumber() {
...
             }
 
-            if (attributeExpressionLength == 3) {
-                if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+
+            if (attributeExpressionLength >= 3) {
+
+                if (!(attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor)) {
+                    throw new SiddhiAppValidationException("windowSize has to be constant but " +
+                            "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
```
Comment (miyurud) [inline modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: here this.attributeExpressionExecutors[1].getClass().getCanonicalName() should be changed to

this.attributeExpressionExecutors[2].getClass().getCanonicalName()

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java [hunk 6]
```diff
@@ -249,18 +273,40 @@ private static int getLogFileSequenceNumber() {
...
+                } else {
                     recordWindow = (int) ((ConstantExpressionExecutor) attributeExpressionExecutors[2]).getValue();
+                }
+            }
+
+            if (attributeExpressionLength == 4) {
+
+                if (!(attributeExpressionExecutors[3] instanceof ConstantExpressionExecutor)) {
+                    throw new SiddhiAppValidationException("ID has to be constant but " +
+                            "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
```
Comment (miyurud) [inline modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: here this.attributeExpressionExecutors[1].getClass().getCanonicalName()); should be changed to

this.attributeExpressionExecutors[3].getClass().getCanonicalName());

Comment (SaranganJ) [reply modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java (hunk 5 above)
Comment (SaranganJ) [reply modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java (hunk 4 above)
Comment (SaranganJ) [reply modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java (hunk 3 above)
Comment (SaranganJ) [reply modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df

File: modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java (hunk 2 above)
Comment (SaranganJ) [reply modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java]: Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df

**Pull request merged**